### PR TITLE
HOSTEDCP-1557: Allow the NSG ID to drive the NSG Resource Group for the Azure Cloud Provider Config

### PR DESCRIPTION
**Pull request by bryan-cox**
**What this PR does / why we need it**:
Sets the network security group (NSG) resource group based on the resource group in the NSG ID.

**Which issue(s) this PR fixes**:
Fixes [HOSTEDCP-1557](https://issues.redhat.com/browse/HOSTEDCP-1557)

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [x] This change includes unit tests.

### CODE REVIEW

File: cmd/infra/azure/create.go [hunk 1]
```diff
@@ -184,7 +184,7 @@ func (o *CreateInfraOptions) Run(ctx context.Context, l logr.Logger) (*CreateInf
 		// Extract network security group name
 		if vnet.Properties.Subnets[0].Properties.NetworkSecurityGroup != nil && vnet.Properties.Subnets[0].Properties.NetworkSecurityGroup.ID != nil {
 			result.SecurityGroupID = *vnet.Properties.Subnets[0].Properties.NetworkSecurityGroup.ID
-			securityGroupName, err := azureutil.GetNetworkSecurityGroupNameFromNetworkSecurityGroupID(*vnet.Properties.Subnets[0].Properties.NetworkSecurityGroup.ID)
+			securityGroupName, _, err := azureutil.GetNSGNameAndRGFromNSGID(*vnet.Properties.Subnets[0].Properties.NetworkSecurityGroup.ID)
```
Comment (stevekuznetsov) [inline cmd/infra/azure/create.go:187]: nit: I get why, but having to learn a bunch of acronyms is not ideal. Maybe `NameAndResourceGroupForNetworkSecurityGroupID`? We write this once, read it many times

Comment (bryan-cox) [reply cmd/infra/azure/create.go:187]: Yeah that works. I try not to put things in acronyms as I can and be verbose instead :)

File: support/azureutil/azureutil_test.go [hunk 2]
```diff
@@ -52,37 +52,42 @@ func TestGetNetworkSecurityGroupNameFromNetworkSecurityGroupID(t *testing.T) {
...
 			g := NewGomegaWithT(t)
-			subnetID, err := GetNetworkSecurityGroupNameFromNetworkSecurityGroupID(tc.nsgID)
+			nsgID, nsgRG, err := GetNameAndResourceGroupFromNetworkSecurityGroupID(tc.nsgID)
 			if tc.expectedErr {
 				g.Expect(err).To(Not(BeNil()))
 				g.Expect(err).To(HaveOccurred(), "invalid nsg ID format: "+tc.nsgID)
 			} else {
 				g.Expect(err).To(BeNil())
-				g.Expect(subnetID).To(Equal(tc.expectedNSGName))
+				g.Expect(nsgID).To(Equal(tc.expectedNSGName))
```
Comment (bennerv) [inline support/azureutil/azureutil_test.go:89]:
```suggestion
				g.Expect(nsgName).To(Equal(tc.expectedNSGName))

File: support/azureutil/azureutil_test.go [hunk 3]
```diff
@@ -52,37 +52,42 @@ func TestGetNetworkSecurityGroupNameFromNetworkSecurityGroupID(t *testing.T) {
...
 			expectedNSGName: "myNSGName",
+			expectedNSGRG:   "myResourceGroupName",
 			expectedErr:     false,
 		},
 	}
 	for _, tc := range tests {
 		t.Run(tc.testCaseName, func(t *testing.T) {
 			g := NewGomegaWithT(t)
-			subnetID, err := GetNetworkSecurityGroupNameFromNetworkSecurityGroupID(tc.nsgID)
+			nsgID, nsgRG, err := GetNameAndResourceGroupFromNetworkSecurityGroupID(tc.nsgID)
```
Comment (bennerv) [inline support/azureutil/azureutil_test.go:83]:
```suggestion
			nsgName, nsgRG, err := GetNameAndResourceGroupFromNetworkSecurityGroupID(tc.nsgID)

**Review by bennerv — COMMENTED** [review]
one minor nit, otherwise lgtm

**Pull request merged**